modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -147,39 +154,41 @@ object CalibrationsService {
           }
         } else none.pure[F]
 
-      private def uniqueConfiguration(pid: Program.Id, selection: ObservationSelection)(using Transaction[F]): F[(List[GmosNConfigs], List[GmosSConfigs])] = {
-        val all: F[List[Config.GmosNorth | Config.GmosSouth]] = services.generatorParamsService.selectAll(pid, selection = selection)
-          .map(_.values.toList.map(_.map(_.observingMode)).collect {
-            case Right(mode: Config.GmosNorth) => mode
-            case Right(mode: Config.GmosSouth) => mode
+      private def uniqueConfiguration(pid: Program.Id, selection: ObservationSelection)(using Transaction[F]): F[(List[(Observation.Id, GmosNConfigs)], List[(Observation.Id, GmosSConfigs)])] = {
+        val all: F[List[(Observation.Id, Config.GmosNorth) | (Observation.Id, Config.GmosSouth)]] = services.generatorParamsService.selectAll(pid, selection = selection)
```

Comment:
I think I asked this before, but will this keep growing as we add more instruments?

Reply:
yes it will